libyul/AssemblyStack.cpp

```diff
@@ -249,21 +249,38 @@ pair<MachineAssemblyObject, MachineAssemblyObject> AssemblyStack::assembleAndGue
 	);
 
 	MachineAssemblyObject runtimeObject;
-	// Heuristic: If there is a single sub-assembly, this is likely the runtime object.
-	if (assembly.numSubs() == 1)
+
+	auto initRuntimeObject = [&](size_t i)
+	{
+			evmasm::Assembly& runtimeAssembly = assembly.sub(i);
+			runtimeObject.bytecode = make_shared<evmasm::LinkerObject>(runtimeAssembly.assemble());
+			runtimeObject.assembly = runtimeAssembly.assemblyString();
+			runtimeObject.sourceMappings = make_unique<string>(
+				evmasm::AssemblyItem::computeSourceMapping(
+					runtimeAssembly.items(),
+					{{scanner().charStream() ? scanner().charStream()->name() : "", 0}}
+				)
+			);
+	};
+
+	// Pick matching runtime assembly if name was given
+	if (_runtimeName.has_value())
 	{
-		evmasm::Assembly& runtimeAssembly = assembly.sub(0);
-		runtimeObject.bytecode = make_shared<evmasm::LinkerObject>(runtimeAssembly.assemble());
-		runtimeObject.assembly = runtimeAssembly.assemblyString();
-		runtimeObject.sourceMappings = make_unique<string>(
-			evmasm::AssemblyItem::computeSourceMapping(
-				runtimeAssembly.items(),
-				{{scanner().charStream() ? scanner().charStream()->name() : "", 0}}
-			)
-		);
+		for (size_t i = 0; i < assembly.numSubs(); i++)
+			if (assembly.name() == _runtimeName)
```

Comment:
```suggestion
			if (assembly.sub(i).name() == _runtimeName)
```

Reply:
This one really confuses me because I distinctively remember fixing this before pushing.. must have forgotten to save or something.. hmm got spotting.